src/main/java/uk/gov/hmcts/divorce/common/service/task/SetDateSubmitted.java

```diff
@@ -0,0 +1,37 @@
+package uk.gov.hmcts.divorce.common.service.task;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.ccd.sdk.api.CaseDetails;
+import uk.gov.hmcts.divorce.divorcecase.model.CaseData;
+import uk.gov.hmcts.divorce.divorcecase.model.State;
+import uk.gov.hmcts.divorce.divorcecase.task.CaseTask;
+
+import java.time.Clock;
+import java.time.LocalDateTime;
+
+import static uk.gov.hmcts.divorce.divorcecase.model.State.AwaitingDocuments;
+import static uk.gov.hmcts.divorce.divorcecase.model.State.Submitted;
+
+@Component
+@Slf4j
+public class SetDateSubmitted implements CaseTask {
+
+    @Autowired
+    private Clock clock;
+
+    @Override
+    public CaseDetails<CaseData, State> apply(final CaseDetails<CaseData, State> caseDetails) {
+
+        final CaseData caseData = caseDetails.getData();
+        final State state = caseDetails.getState();
+
+        if (Submitted.equals(state) || AwaitingDocuments.equals(state)) {
```

Comment:
I think we can add HWF here

Reply:
Updated in https://github.com/hmcts/nfdiv-case-api/pull/571/commits/3f45324265a75c89a817a5cbb3e14219d7414d19